diff -ur fsl.orig/config/common/vars.mk fsl/config/common/vars.mk
--- fsl.orig/config/common/vars.mk	2017-07-06 14:02:03.144380856 -0500
+++ fsl/config/common/vars.mk	2017-07-07 17:04:07.424276484 -0500
@@ -24,14 +24,14 @@
 USRCFLAGS = 
 USRCXXFLAGS =
 
-LDFLAGS = ${ARCHLDFLAGS} ${USRLDFLAGS} -L. -L${DEVLIBDIR} -L${LIBDIR}
+LDFLAGS = ${EBVARLDFLAGS} ${ARCHLDFLAGS} ${USRLDFLAGS} -L. -L${DEVLIBDIR} -L${LIBDIR}
 
-AccumulatedIncFlags = -I${INC_BOOST} ${USRINCFLAGS} -I. -I${DEVINCDIR} -I${INCDIR}
+AccumulatedIncFlags = ${EBVARCPPFLAGS} -I${INC_BOOST} ${USRINCFLAGS} -I. -I${DEVINCDIR} -I${INCDIR}
 
-CFLAGS = ${ANSI_FLAGS} ${ANSI_CFLAGS} ${DBGFLAGS} ${USEDCSTATICFLAGS} ${USRCFLAGS} ${ARCHFLAGS} ${OPTFLAGS}  \
+CFLAGS = ${EBVARCFLAGS} ${ANSI_FLAGS} ${ANSI_CFLAGS} ${DBGFLAGS} ${USEDCSTATICFLAGS} ${USRCFLAGS} ${ARCHFLAGS} ${OPTFLAGS}  \
 	${AccumulatedIncFlags}
 
-CXXFLAGS = ${ANSI_FLAGS} ${ANSI_CXXFLAGS} ${DBGFLAGS} ${USEDCXXSTATICFLAGS} ${USRCXXFLAGS} ${ARCHFLAGS} ${OPTFLAGS}  \
+CXXFLAGS = ${EBVARCXXFLAGS} ${ANSI_FLAGS} ${ANSI_CXXFLAGS} ${DBGFLAGS} ${USEDCXXSTATICFLAGS} ${USRCXXFLAGS} ${ARCHFLAGS} ${OPTFLAGS}  \
 	${AccumulatedIncFlags}
 
 HFILES = *.h
diff -ur fsl.orig/config/generic/externallibs.mk fsl/config/generic/externallibs.mk
--- fsl.orig/config/generic/externallibs.mk	2017-07-06 14:02:03.144380856 -0500
+++ fsl/config/generic/externallibs.mk	2017-07-10 10:17:27.720024683 -0500
@@ -6,25 +6,21 @@
 FSLEXTINC=${FSLDIR}/extras/include
 FSLEXTBIN=${FSLDIR}/extras/bin
 
-# CEPHES library
-LIB_CEPHES = ${FSLEXTLIB}
-INC_CEPHES = ${FSLEXTINC}/cephes
-
 # GD library
-LIB_GD = ${FSLEXTLIB}
-INC_GD = ${FSLEXTINC}
+LIB_GD = ${EBROOTLIBGD}/lib
+INC_GD = ${EBROOTLIBGD}/include
 
 # GDC library
 LIB_GDC = ${FSLEXTLIB}
 INC_GDC = ${FSLEXTINC}/libgdc
 
 # GSL library
-LIB_GSL = ${FSLEXTLIB}
-INC_GSL = ${FSLEXTINC}/gsl
+LIB_GSL = ${EBROOTGSL}/lib
+INC_GSL = ${EBROOTGSL}/include
 
 # PNG library
-LIB_PNG = ${FSLEXTLIB}
-INC_PNG = ${FSLEXTINC}
+LIB_PNG = ${EBROOTLIBPNG}/lib
+INC_PNG = ${EBROOTLIBPNG}/include
 
 # PROB library
 LIB_PROB = ${FSLEXTLIB}
@@ -43,24 +39,36 @@
 INC_NEWRAN = ${FSLEXTINC}/newran
 
 # ZLIB library
-LIB_ZLIB = ${FSLEXTLIB}
-INC_ZLIB = ${FSLEXTINC}
+LIB_ZLIB =  ${EBROOTZLIB}/lib
+INC_ZLIB = ${EBROOTZLIB}/include
 
 # QT library
-QTDIR = /usr/lib/qt3
+QTDIR = ${EBROOTQT}
 LIB_QT = ${QTDIR}/lib
 INC_QT = ${QTDIR}/include
 
 # BOOST library
-BOOSTDIR = ${FSLEXTINC}/boost
-LIB_BOOST = ${BOOSTDIR}
-INC_BOOST = ${BOOSTDIR}
+BOOSTDIR = ${EBROOTBOOST}
+LIB_BOOST = ${BOOSTDIR}/lib
+INC_BOOST = ${BOOSTDIR}/include
 
 # QWT library
-QWTDIR = /usr/local/qwt
+QWTDIR = ${EBROOTQWT}
 INC_QWT = ${QWTDIR}/include
 LIB_QWT = ${QWTDIR}/lib
- 
+
 # FFTW3 library
-LIB_FFTW3 = ${FSLEXTLIB}
-INC_FFTW3 = ${FSLEXTINC}/fftw3
+LIB_FFTW3 = ${EBVARFFTW_LIB_DIR}
+INC_FFTW3 = ${EBVARFFTW_INC_DIR}
+
+# LIBXML2 library
+INC_XML2 = ${EBROOTLIBXML2}
+
+# LIBXML++ library
+INC_XML++ = ${FSLEXTINC}/libxml++-2.6
+INC_XML++CONF = ${FSLEXTLIB}/libxml++-2.6/include
+
+# VTK library
+VTKDIR_INC = ${EBROOTVTK}/include/vtk-`echo ${EBVERSIONVTK} | cut -f1-2 -d.`
+VTKDIR_LIB = ${EBROOTVTK}/lib
+VTKSUFFIX = -`echo ${EBVERSIONVTK} | cut -f1-2 -d.`
diff -ur fsl.orig/config/generic/systemvars.mk fsl/config/generic/systemvars.mk
--- fsl.orig/config/generic/systemvars.mk	2017-07-06 14:02:03.144380856 -0500
+++ fsl/config/generic/systemvars.mk	2017-07-10 14:08:28.076009819 -0500
@@ -4,30 +4,32 @@
 
 # System dependent commands (NB: the first two are the most platform dependent)
 
-INSTALL = ginstall -p
-RANLIB = ranlib
+INSTALL = install -p
+RANLIB = echo
 
 RM = /bin/rm
 CP = /bin/cp
 MV = /bin/mv
 CHMOD = /bin/chmod
 MKDIR = /bin/mkdir
-TCLSH = ${FSLDIR}/bin/fsltclsh
+TCLSH = tclsh
+FSLML = ${FSLDIR}/bin/fslml
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
+CXX11 = ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
-ARCHFLAGS = 
-
+ARCHFLAGS = -m64
+PARALLELFLAGS = -fopenmp
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS}
-MACHDBGFLAGS =
-GNU_ANSI_FLAGS = -Wall -ansi -pedantic
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS}
+MACHDBGFLAGS = -g
+GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
 ANSI_FLAGS = ${GNU_ANSI_FLAGS}
 
diff -ur fsl.orig/src/mist-clean/Makefile fsl/src/mist-clean/Makefile
--- fsl.orig/src/mist-clean/Makefile	2017-07-06 14:02:03.096380856 -0500
+++ fsl/src/mist-clean/Makefile	2017-07-10 11:53:41.356018491 -0500
@@ -1,16 +1,16 @@
 include ${FSLCONFDIR}/default.mk
 
-NLOPT_INC = ${FSLEXTINC}
-NLOPT_LIB = ${FSLEXTLIB}
-SQLITE_INC = ${FSLEXTINC}/libsqlite
-SQLITE_LIB = ${FSLEXTLIB}
+NLOPT_INC = ${EBROOTNLOPT}/include
+NLOPT_LIB = ${EBROOTNLOPT}/lib
+SQLITE_INC = ${EBROOTSQLITE}/include
+SQLITE_LIB = ${EBROOTSQLITE}/lib
 
 PROJNAME = mist
 
 XFILES = mist/mist
 SCRIPTS = bin/mist_1_train bin/mist_2_fit bin/mist_FA_reg bin/mist_display bin/mist_mesh_utils
 
-USRCXXFLAGS = -std=c++11
+USRCXXFLAGS = -std=c++11 -DBOOST_LOG_DYN_LINK
 USRINCFLAGS = -I${FSLDIR}/include/newimage -I${INC_NEWMAT} -I${INC_ZLIB} -I${INC_GDC} -I${INC_GD} -I${SQLITE_INC} -I${NLOPT_INC} -I${VTKDIR_INC} -Icommon
 USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_ZLIB} -L${LIB_BOOST} -L${LIB_GDC} -L${LIB_GD} -L${NLOPT_LIB} -L${VTKDIR_LIB}
 
